static/app/views/relocation/relocation.tsx

```diff
@@ -56,49 +59,103 @@ function getRelocationOnboardingSteps(): StepDescriptor[] {
       Component: UploadBackup,
       cornerVariant: 'top-left',
     },
+    {
+      id: 'in-progress',
+      title: t('Your relocation is in progress'),
+      Component: InProgress,
+      cornerVariant: 'top-left',
+    },
   ];
 }
 
+enum LoadingState {
+  FETCHED,
+  FETCHING,
+  ERROR,
+}
+
 function RelocationOnboarding(props: Props) {
-  const [hasPublicKeyError, setHasError] = useState(false);
+  const {
+    params: {step: stepId},
+  } = props;
+  const onboardingSteps = getRelocationOnboardingSteps();
+  const stepObj = onboardingSteps.find(({id}) => stepId === id);
+  const stepIndex = onboardingSteps.findIndex(({id}) => stepId === id);
+  const api = useApi();
+  const regions = ConfigStore.get('regions');
+
+  const [existingRelocationState, setExistingRelocationState] = useState(
+    LoadingState.FETCHING
+  );
+  const [existingRelocation, setExistingRelocation] = useState('');
 
   // TODO(getsentry/team-ospo#214): We should use sessionStorage to track this, since it should not
   // change during a single run through this workflow.
   const [publicKey, setPublicKey] = useState('');
+  const [publicKeyState, setPublicKeyState] = useState(LoadingState.FETCHING);
+
+  const fetchExistingRelocation = useCallback(() => {
+    setExistingRelocationState(LoadingState.FETCHING);
+    return Promise.all(
```

Comment:
maybe you're looking for Promise.race if you just want the first response?

Reply:
I don't want the first response - I want to query every active region, then make a decision based on whether or not there is an active relocation in any of them. So we need wait until all of the promises resolve.